File: Programmers/JeongHyun/4주차 (정렬)/K번째수.js
function solution(array, commands) {
    var answer = [];
    for(let i=0;i<commands.length;++i){
        let arr = [];
        /*
        commands 
        commands[i][0]-1 = 1
        commands[i][1] = 5
        commands[i][2]-1 = 2
        */ 
        for(let j=commands[i][0]-1;j<commands[i][1];++j){
            arr.push(array[j]);
        }
        arr.sort((a,b)=>a-b);
        answer.push(arr[commands[i][2]-1]);
    }
    return answer;
}


function solution(array, commands) {
    var answer = [];
    answer = commands.map(a=>{
        // map은 commands를 이용하여 answer을 만들어주는 함수로서
        // a[0]-1 을 하면 commands[i][0] - 1로 범위만큼 array를 자릅니다.
        // .sort를 통해 정렬한 뒤 a[2]-1 만을 리턴해줍니다. 이 과정을 commands의 length만큼 진행합니다.
        return array.slice(a[0]-1,a[1]).sort((b,c)=>b-c)[a[2]-1];
    })
    return answer;
}

File: Programmers/SeungDo/4주차 - 정렬/K번째 수.cpp
#include <string>
#include <vector>
#include<algorithm>
using namespace std;
// comand에 들어가있는 데이터 갯수 만큼 반복문을 돌려 i번째부터 j번째까지 
// array에서 자른후 오름차순 정렬해 k번째 숫자를 answer에 push하는 방식으로 했습니다.
vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for (int i = 0; i < commands.size(); i++) {
        vector <int> selection; // array에서 자른 숫자들이 들어가는 배열 
        for (int j = commands[i][0] - 1; j <= commands[i][1] - 1; j++)  selection.push_back(array[j]); //숫자 자르기
        sort(selection.begin(), selection.end()); // 오름차순 정렬
        answer.push_back(selection[commands[i][2] - 1]); // k번째 수 answer에 푸쉬
    }
    return answer;
}
